mathesar/api/db/viewsets/columns.py

```diff
@@ -31,6 +32,9 @@ def get_queryset(self):
     def create(self, request, table_pk=None):
         table = get_table_or_404(table_pk)
         # We only support adding a single column through the API.
+        if(request.data['name'].strip() == ''):
+            request.data['name'] = gen_col_name(table)
```

Comment:
This is not where you should be including the column generation logic. Columns are actually created in the `create_column` method in `db/columns/operations/create.py`, that's where you should generate the name if none is passed to it.